pulp_smash/tests/docker/cli/test_copy.py

```diff
@@ -0,0 +1,244 @@
+        self.assertEqual(self.completed_proc.returncode, 0)
+
+    def test_task_succeeded(self):
+        """Assert that "Copied" appears in stdout."""
+        self.assertIn('Copied', self.completed_proc.stdout)
+
+
+class CopyAllManifestsTestCase(docker_utils.BaseTestCase,
+                               docker_utils.SuccessMixin):
+    """Test copying all Manifests from one repository to another."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create and sync a docker repository with a v2 registry."""
+        super(CopyAllManifestsTestCase, cls).setUpClass()
+        if cls.cfg.version < Version('2.8'):
+            raise unittest2.SkipTest('These tests require Pulp 2.8 or above.')
+        # Create and sync a repository with a v2 feed to bring some Manifests
+        # in.
+        docker_utils.create_repo(
+            cls.cfg, cls.repo_id, upstream_name='library/busybox',
+            sync_v1=False, sync_v2=True)
+        docker_utils.sync_repo(cls.cfg, cls.repo_id)
+
+        # Now create a feedless repository and copy the manifests from the
+        # first repository over.
+        cls.copy_target = utils.uuid4()
+        docker_utils.create_repo(cls.cfg, cls.copy_target)
+        cls.completed_proc = docker_utils.copy(cls.cfg, 'manifest',
+                                               cls.repo_id, cls.copy_target)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the copy_target repo."""
+        super(CopyAllManifestsTestCase, cls).tearDownClass()
+        docker_utils.delete_repo(cls.cfg, cls.copy_target)
+
+    def test_positive_copy_output(self):
+        """Assert that the list of manifests in the copy output are correct."""
+        units_in_src = docker_utils.search(self.cfg, 'manifest', self.repo_id,
+                                           fields=['digest']).stdout
+        # Due to https://pulp.plan.io/issues/1693 it is not possible to get
+        # pulp-admin to limit the output to the digests. Hello regex my old
+        # friend, I've come to talk with you again…
+        manifest_ids_in_src = set(re.findall(r'(?:Digest:)\s*(.*)',
+                                             units_in_src))
+
+        # The manifest digests are printed after "docker_manifest:" in the copy
+        # output.
+        manifest_ids = self.completed_proc.stdout.split('docker_manifest:')[1]
+        manifest_ids_printed = re.findall(r'(?: {2})(\w*:\w*)', manifest_ids)
+        # Due to https://pulp.plan.io/issues/1696 pulp-admin has split the last
+        # character of each of the manifest_ids_in_src items onto the next line
+        # which wasn't matched by the regex. That's stupid, but we're going to
+        # work around it by chopping the last character off of these as well.
+        manifest_ids_printed = set([d[:-1] for d in manifest_ids_printed])
+
+        self.assertEqual(manifest_ids_printed, manifest_ids_in_src)
+
+    def test_positive_units_copied(self):
```

Comment:
They aren't actually identical, and the SuccessMixin isn't specific to copying. I don't think we need to worry about their similarities, they do work.